splunklib/client.py

```diff
@@ -463,6 +463,13 @@ def info(self):
         response = self.get("/services/server/info")
         return _filter_content(_load_atom(response, MATCH_ENTRY_CONTENT))
 
+    def input(self, path, kind=None):
+        """Retrieves an input by path, and optionally kind.
+
+        :return: A :class:`Input` object.
+        """
+        return Input(self, path, kind=kind).refresh()
```

Comment:
**Non-blocking Personal Opinion**: I like to see these operations broken down atomically because it makes things easier to debug.

```
new_input = Input(self, path, kind=kind)
new_input.refresh()
return new_input
```

Reply:
fair, I'm just following the style I used for `job()`